### ISSUE TITLE
Use temporary directories

### ISSUE THREAD

**Issue by spelunk** [faressoft/terminalizer#29]
```
$ terminalizer render foobar
Error: 
  EACCES: permission denied, mkdir '/usr/local/lib/node_modules/terminalizer/render/frames'
```

Skimming the code, [places hardcode the relative `render/frames/` directory](https://github.com/faressoft/terminalizer/blob/master/commands/render.js#L197).  These probably should be stored in a temporary directory using a module like [`tempfile`](
https://www.npmjs.com/package/tempfile)

**Comment by faressoft** [comment]
@spelunk I will consider storing the files in the temporary directory

For now you can solve the issue by fixing the npm permissions

```
chown -R faressoft /usr/local/lib/node_modules
chown -R faressoft /usr/local/bin
chown -R faressoft /usr/local/share
```

Change `faressoft` to your username.

**Comment by David-OConnor** [comment]
Fix doesn't work for me.

**Comment by paralllax** [comment]
If anyone still needs a work around, this worked for me.

```
sudo chown -R <user> /usr/lib/node_modules/terminalizer/render/
sudo chmod 4775 /usr/lib/node_modules/terminalizer/node_modules/electron/dist/chrome-sandbox
```

**Comment by rsalmei** [comment]
This is very bad. Using 0.7.2 and still doesn't work.

**Comment by volcan01010** [comment]
Using Ubuntu 18.04, I got Terminalizer to work as follows:

```bash
# Install npm (requires dependencies)
sudo apt install npm node-gyp nodejs-dev libssl1.0-dev

# Install terminalizer globally
sudo npm install -g terminalizer

# Remove the electron that comes with terminalizer
sudo rm -rf /usr/local/lib/node_modules/terminalizer/node_modules/electron/

# Add electron globally - based on this SO answer https://stackoverflow.com/a/52033822/3508733
sudo npm install -g electron --unsafe-perm=true --allow-root

# By this point, I could record and play sessions, but got the error in this issue when I tried to render them.
# Symlink the hardcoded rendering path to where terminalizer is actually installed on Ubuntu
sudo mkdir -p /usr/lib/node_modules/
sudo ln -s /usr/local/lib/node_modules/terminalizer/ /usr/lib/node_modules/terminalizer

# Do the workaround above
sudo chown -R <user> /usr/lib/node_modules/terminalizer/render
sudo chmod 4775 /usr/local/lib/node_modules/electron/dist/chrome-sandbox 
```

Thanks @faressoft and @paralllax!

**Comment by Licenser** [comment]
+1 to fixing this, it sadly makes it super hard to use terminalizer on a linux system

**Comment by nick4fake** [comment]
@faressoft are you serious? Everyone, never do something like that.

As for this issue tools should not try to write anything to library paths. chmod solutions is something temporary before there is a real temp directory fix.

As another temporary solution we could symlink /usr/lib/node_modules/terminalizer/render/ to /tmp or some user folder (once again, this still is not something npm users should really do).

**Comment by xsebek** [comment]
@faressoft could you please at least handle the issue gracefully with a message like:
```
Error:
  Missing access to global installation folder ('/usr/local/lib/node_modules/terminalizer/render/frames').
  Please see https://github.com/faressoft/terminalizer/issues/29 for a workaround.
```
I would like to use `terminalizer` globally, so local installation where permissions are correct (?) does not work for me.

The GIF rendering is the main reason I want to use `terminalizer` and this makes it needlessly harder. :disappointed:

**Comment by festum** [comment]
Why I'm still facing this issue it's 2023!

**Comment by faressoft** [comment]
Fixed in #226

**Comment by kwk** [comment]
@faressoft I still face this issue even with 0.12.0:

```console
$ terminalizer --version
0.12.0
$ terminalizer render demo.yml
Error: 
  Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/terminalizer/render/frames'

Hint:
  Use the --help option to get help about the usage
```
